src/redux/features/Plantings/plantingsSelectors.ts

```diff
@@ -9,3 +11,25 @@ export const selectPlantingsDateRange = (state: RootState, dateRange: string[])
     }
     return planting.createdTime > dateRange[0] && (dateRange.length < 2 || planting.createdTime < dateRange[1]);
   }) ?? [];
+
+export const selectPlantingProgressSubzones = createSelector(
+  [
+    (state: RootState, plantingSiteId: number) => selectPlantingSite(state, plantingSiteId),
```

Comment:
The selector looks fine for a single site.
thoughts on supporting all sites, searching by name and or list of ids, etc?
I think that may influence your design of selectors with reusable functions.

Reply:
True, we need all sites. Will try to improve this

Reply:
Thanks, that will be great! if you need help lmk.

Reply:
Here's a suggestion if it helps:

```
// selector for all sites,
const getReportedPlants = createSelector(
[
   (state) => selectPlantingSites,
    (state) => selectPlantings,
], (sites, plantings) => {
   return sites?.map((site, plantings) => {
    return {
      siteId: site.id,
      siteName: site.name,
      reported:     .. do what you are currently doing , so we get an array of results for every site 
   };
  }
})

// selector for single site
const getReportedPlantsForSite = (state, siteId) => getReportedPlants(state)?.find(report => report.siteId === siteId);

// for search and filtering
const searchReportedPlants = (state, siteIds: string[], query: string) => same as for single site but filter instead of find
```